core/cli/src/main/java/mil/nga/giat/geowave/core/cli/operations/config/SetCommand.java

```diff
@@ -46,44 +43,68 @@
 	@Parameter(names = {
 		"--password"
 	}, description = "boolean (true|false) - specify if the value being set is a password and should be encrypted in the configurations")
-	private String password = null;
+	private final String password = null;
 
 	private boolean isPassword;
 
 	@Override
-	public boolean prepare(
-			OperationParams params ) {
-		super.prepare(params);
-		if (password != null && !"".equals(password.trim())) {
+	public void execute(
+			final OperationParams params ) {
+		if ((password != null) && !"".equals(password.trim())) {
 			isPassword = Boolean.parseBoolean(password.trim());
 		}
-		return true;
+		computeResults(params);
 	}
 
+	/**
+	 * Add rest endpoint for the set command. Looks for GET params with keys
+	 * 'key' and 'value' to set.
+	 *
+	 * @return string containing json with details of success or failure of the
+	 *         set
+	 */
 	@Override
-	public void execute(
-			OperationParams params ) {
-
-		Properties existingProps = getGeoWaveConfigProperties(params);
+	public Object computeResults(
+			final OperationParams params ) {
+		try {
+			return setKeyValue(params);
+		}
+		catch (WritePropertiesException | ParameterException e) {
+			// TODO GEOWAVE-rest-project server error status message
+			// this.setStatus(
+			// Status.SERVER_ERROR_INTERNAL,
+			// e.getMessage());
+			return null;
+		}
+	}
 
-		PasswordConverter converter = new PasswordConverter(
+	/**
+	 * Set the key value pair in the config. Store the previous value of the key
+	 * in prevValue
+	 */
+	private Object setKeyValue(
+			final OperationParams params ) {
+
+		final File f = (File) params.getContext().get(
+				ConfigOptions.PROPERTIES_FILE_CONTEXT);
+		final Properties p = ConfigOptions.loadProperties(
+				f,
 				null);
 
 		String key = null;
 		String value = null;
-
-		if (parameters.size() == 1 && parameters.get(
+		if ((parameters.size() == 1) && (parameters.get(
 				0).indexOf(
-				"=") != -1) {
-			String[] parts = StringUtils.split(
+				"=") != -1)) {
+			final String[] parts = StringUtils.split(
 					parameters.get(0),
 					"=");
 			key = parts[0];
-			value = converter.convert(parts[1]);
+			value = parts[1];
```

Comment:
we got rid of password converter, seems like a discrepancy with the old code

Reply:
when execute() call this set a flag to indicate we still want to use the password converter, except only use the password converter if isPassword

Reply:
Done